### PR TITLE
iOS Support ; Latest Xamarin.Forms ; Sample App updates.

### PR DESCRIPTION
**Pull request by SwiftArchitect**
#23 
Keep up to date with Xamarin.Forms
Tested with Visual Studio for Mac Community 7.4.2 build 12, and on Android device.

#15 
Added some support for iOS. Includes functional iOS Sample App

### CODE REVIEW

**Comment by Olivier-B** [comment]
Hi,

Just had a talk to SwiftArchitect, and waiting for "commit" of his branche for iOS component.

Olivier

**Comment by andijakl** [comment]
Quick question: you switched the target framework of the core library from .net standard 1.3 to 2.0. Is the higher version actually required by the code changes? Staying at .net standard 1.3 has the benefit of increased compatibility (e.g., to .net core 1.0+ instead of just 2.0+). If newer APIs are not required, I think it's best for a library to maximize compatibility, as this shouldn't have any other limitations or side effects?

**Comment by andijakl** [comment]
Another question we should discuss: you chose the MIT license for the iOS part. The rest of the library is under Apache 2.0 license (the examples under GPLv3). I wouldn't like to mix 3 licenses in one project.
In general, I'd have no problem changing the overall license from Apache to MIT. But not sure if that's so easy for projects that are using the library, as they'd have to re-evaluate. So the easier variant would be to also make the iOS part Apache 2.0. Or do you have other ideas?

**Comment by andijakl** [comment]
When testing the current complete code, I found that the Android Xamarin app didn't have the run-time permission request implemented yet. It's required on recent Android versions to scan for Bluetooth LE devices.

I've added the permission for Android to the Manifest and also added a run-time permission request through the Xamarin Permissions Plugin: https://github.com/jamesmontemagno/PermissionsPlugin

The plug-in has the disadvantage that it can only request the fine + course location, even though "fine" wouldn't be required ("course" location is enough for beacon scanning). The advantage of using the plug-in is that it'd also work for iOS, where I also didn't see the permission request implemented. I don't have a mac / iPhone to test - but from what I read, checking at run-time for permissions is also needed on iOS? Can you please check & test that?

There's also still an issue on Android when I tested on the Pixel 2 with Android 9. It does discover beacons, but only prints out messages like this:
[0:] Provider_AdvertisementPacketReceived md51d86cc97c1695ccd43174683b3b4883e.AndroidBluetoothPacketProvider@9b51d83 Beacon UniversalBeacon.Library.Core.Interop.BLEAdvertisementPacketArgs
There's no visual info added to the UI. Need to investigate what's wrong here. Would also be happy if anyone else has an idea :)

**Comment by SwiftArchitect** [comment]
I am not concerned about compatibility with the past but with the future:
by using 2.0, it is my understanding that you are more in line with
requirements from other NuGet, leading to more robust and leaner binaries.

If you have developers limited to 1.3, then they can use the current
version of the NuGet.

Going forward, I would like to be in line with the latest versions of any
libraries and other dependencies. Especially on edge technologies like
Bluetooth, you probably want to rely upon the latest advancements in OS
behavior and security.

I wouldn't go any lower than what the Mac OS Xamarin requires.

**Comment by ctacke** [comment]
Two things:

1. Yes, I strongly prefer MIT.  It's lightweight, very permissible, and is as close to zero friction as I've seen by enterprises looking at license restrictions.  Anything stamped with GPL I largely won't even look at, and many, many enterprises are the same way.

2. Net Standard 1.3 is fully consumable by Net Standard 2.0.  Broadly speaking when building a library, you should compile against ever smaller version numbers until it breaks you and go with the lowest number possible to get the broadest possible usability surface area.  Targeting 2.0 just because it's newer  is not the way to go.  Generally anything 1.2 and earlier is really, really API limited.  If we can still compile against 1.3 with the new additions, then we should stay at 1.3.  It won't affect new consumers whose apps use 2.x and it won't break people currently using or wishing to target 1.x.

File: Library/UniversalBeacon.Library.Cocoa/CocoaBluetoothPacketProvider.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        public void Start()
+        {
+            Debug.WriteLine("BluetoothPacketProvider:Start()");
+            centralDelegate.OnAdvertisementPacketReceived += ScanCallback_OnAdvertisementPacketReceived;
+
+            // Wait for the PoweredOn state
+
+            //if(CBCentralManagerState.PoweredOn == central.State) {
```
[the remark this answers is not in the document]
Comment (ctacke) [reply Library/UniversalBeacon.Library.Cocoa/CocoaBluetoothPacketProvider.cs]: Understood.  I'm just not a fan of block commented code and whenever I see it, it needs to at least be reasonably justified.  My preference would be for the block to not get merged into the master branch, but I also understand that development is messy and if it helps you with your work, I'm not going to reject it.  Carry on, and thanks for your efforts!  It's now one of those "things I've been meaning to do myself" that I now don't have to.

**Comment by SwiftArchitect** [comment]
@ctacke 
I'm all for 1.3 Net Standard, MIT and _merge_ if at all possible since I do not have the write access.
Can you fix the 1.3 or should I update the PR?

If deemed acceptable, I can address the commented out code in the upcoming PR for https://github.com/andijakl/universal-beacon/issues/27

**Pull request merged**